.github/workflows/run-mypy-on-modified-cdk-files.yml

```diff
@@ -0,0 +1,30 @@
+name: Run mymy on modified cdk files
```

Comment:
s/mymy/mypy :)

Also, there might be a naming convention these days for gha, check with @airbytehq/connector-operations